### PR TITLE
Render patternProperties

### PR DESCRIPTION
**Pull request by arnulfojr**
### Fixes:
https://github.com/aws-cloudformation/cloudformation-cli/issues/418

### Description of changes
Allow patternProperties to be documented and not failing the doc generation if no properties were found.
Escape propname in markdown (in case of patterProperty)

I created this PR to trigger a discussion for the issue linked, looking forward to read your comments!

> I was using Python 3.7 and seems that the pre-commit isort and black are contradicting each other

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/rpdk/core/project.py [hunk 1]
```diff
@@ -432,10 +447,20 @@ def _set_docs_properties(self, propname, prop, proppath):
...
             docs_path = self.root / "docs"
 
-            prop["properties"] = {
-                name: self._set_docs_properties(name, value, proppath + (name,))
-                for name, value in prop["properties"].items()
-            }
+            if "properties" in prop:
+                prop["properties"] = {
+                    name: self._set_docs_properties(name, value, proppath + (name,))
+                    for name, value in prop["properties"].items()
```
Comment (jotompki) [inline src/rpdk/core/project.py:453]:
```
try:
    prop["properties"] = {
                    name: self._set_docs_properties(name, value, proppath + (name,))
                    for name, value in prop.get("properties", prop["patternProperties"])
                }
except KeyError:
    raise InvalidSchemaPropertyError(
                    "Object property require a properties or patternProperties field"
                )
```
 could consolidate this logic nicely without all the branching.  i don't know if we need to even worry about a schema where neither properties or patternproperties is specified, but it guess it doesn't hurt including it

Comment (arnulfojr) [reply src/rpdk/core/project.py:453]: The problem with the solution is that `prop["patternProperties"]` will be evaluated before dict fallback happens so the KeyError always happens.

This is what would be in JS
```js
prop.get("properties", () => prop["patternProperties"])
```

Possibile solutions:
```python
            try:
                object_properties = prop["properties"]
                  if "properties" in prop else prop["patternProperties"]
            except KeyError:
                raise InvalidSchemaPropertyError(
                    "Object property requires a properties or patternProperties field"
                )

            prop["properties"] = {
                name: self._set_docs_properties(name, value, proppath + (name,))
                for name, value in object_properties.items()
            }
```

```python
            try:
                prop["properties"] = {
                    name: self._set_docs_properties(name, value, proppath + (name,))
                    for name, value in prop.get(
                        "properties", prop.get("patternProperties")
                    ).items()
                }
            except AttributeError:
                raise InvalidSchemaPropertyError(
                    "Object property requires a properties or patternProperties field"
                )

```

or the above with default to empty list

Comment (arnulfojr) [reply src/rpdk/core/project.py:453]: I pushed the first possible solution, let me know what you think :)

**Comment by jotompki** [comment]
Looks good

File: src/rpdk/core/project.py [hunk 2]
```diff
@@ -432,9 +449,20 @@ def _set_docs_properties(self, propname, prop, proppath):
             template = self.env.get_template("docs-subproperty.md")
             docs_path = self.root / "docs"
 
+            try:
+                object_properties = (
```
Comment (wbingli) [inline src/rpdk/core/project.py:453]: The `object` is valid is without `properties` or `patternProperties`.  See [SNS schema](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-sns/blob/master/aws-sns-topicpolicy/aws-sns-topicpolicy.json#L9). 
Could you help to address this?

Comment (arnulfojr) [reply src/rpdk/core/project.py]: I see! I didn't know that was possible, now I wonder how CDK handles this 🤔
It seems to be that they have it there as a `any` TS type for the L1 and added some types for the L2... Interesting

Anyways, I pushed new rev for the PR

**Pull request merged**